### DISCUSSION TITLE
Welcome to Edge-Detect Discussions!

### DISCUSSION THREAD

**Discussion by jishnujayakumar** [racsa-lab/Edge-Detect#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jfreeman7200** [comment]
Good morning!  Are you still working on this project?

**Reply by jishnujayakumar** [reply]
Currently, no.

**Comment by freeEDUfi** [comment]
Hi , can you please give me a clear instruction what python file I should run? I see generator and recurrent_fast file I got index out of range in generator error the error is in timestep = int(sys.argv[1]) line


timestep = int(sys.argv[1])
print(timestep)
# load dataset
dataset = pd.read_csv('training.csv', header=0)
dataset = dataset.dropna()
dataset = dataset.fillna(method="ffill")
dataset = dataset.apply(pd.to_numeric, errors="coerce")
train_y = dataset.values[:, -1]
orgshape = dataset.shape[1]
columns = [dataset.shift(i) for i in range(timestep, 0, -1)]
dataset = pd.concat(columns, axis=1)
values = dataset.values.reshape(dataset.shape[0], timestep, orgshape)

for i in range(timestep):
    values = np.delete(values, 0, axis=0)
    train_y = np.delete(train_y, 0, axis=0)

# split into train and test sets

train = values[:, :, :]
# split into input and outputs
train_X = train[:, :, :-1]

dataset = pd.read_csv('testing.csv', header=0)
dataset = dataset.dropna()
dataset = dataset.fillna(method="ffill")
dataset = dataset.apply(pd.to_numeric, errors="coerce")
test_y = dataset.values[:, -1]
columns = [dataset.shift(i) for i in range(timestep, 0, -1)]
dataset = pd.concat(columns, axis=1)
values = dataset.values.reshape(dataset.shape[0], timestep, orgshape)
for i in range(timestep):
    values = np.delete(values, 0, axis=0)
    test_y = np.delete(test_y, 0, axis=0)
# integer encode direction

test = values[:, :, :]
# split into input and outputs


test_X = test[:, :, :-1]
# reshape input to be 3D [samples, timesteps, features]

print(train_X.shape, train_y.shape, test_X.shape, test_y.shape)

np.save("TRAINING_X_" + str(timestep) + ".npy", train_X)
np.save("TRAINING_Y_" + str(timestep) + ".npy", train_y)

np.save("TESTING_X_" + str(timestep) + ".npy", test_X)
np.save("TESTING_Y_" + str(timestep) + ".npy", test_y)

**Reply by 2358815210** [reply]
Is your code running?

**Reply by jishnujayakumar** [reply]
Please modify `training/pi_queue.py` to do the training. The test and train both have the same code by accident.
Also, `recurrent_fast.py` has the LSTM and GRU classes with `FastRNN` and `FastGRNN` capabilities. You would need to modify the installed TensorFlow library core as per it.

**Reply by 2358815210** [reply]
Which version does TensorFlow use?How to modify parameters in"training/pi_queue.py"

**Reply by jishnujayakumar** [reply]
Refer https://www.tensorflow.org/api_docs/python/tf/keras/Model to use a keras model and use one from all_models.py

Kindly check what works best for you (tf1 / tf2).

**Reply by 2358815210** [reply]
Can you share the "environment.yml" of this project
Do you need to assign value to "timestep = int(sys.argv[1]) "

**Reply by codelics** [reply]
> Please modify `training/pi_queue.py` to do the training. The test and train both have the same code by accident. Also, `recurrent_fast.py` has the LSTM and GRU classes with `FastRNN` and `FastGRNN` capabilities. You would need to modify the installed TensorFlow library core as per it.

could you please explain the main purpose of the recurrent_fast.py, it seems like it's not connected to other python files.  and is it possible to get the user input and predict it using this kind of model?  I had a problem in testing.py so I don't have an idea what to do with the model. load_weights because it's not working and it seems its asking for an h5py file, so what I tried is to save weights from the training.py as h5py file  but I am not sure if thats the right thing to do.

**Reply by codelics** [reply]
> Please modify `training/pi_queue.py` to do the training. The test and train both have the same code by accident. Also, `recurrent_fast.py` has the LSTM and GRU classes with `FastRNN` and `FastGRNN` capabilities. You would need to modify the installed TensorFlow library core as per it.

could you please specify which part of training.py I should modify? am I supposed to remove some line of codes there? or just add some codes for the training?

**Comment by 2358815210** [comment]
Which version does tensorflow need?

**Comment by joshuacbngn** [comment]
Hello! Can you elaborate on the pre-processing stage for the dataset? The files I downloaded from the project had 80 features (the last feature being the labels for attacks), but I read in the paper that it was summarized from an initial 49 features to 25. How can I achieve the same dataset? I've tried running the project using the training.csv file and testing.csv file but those appear to have 80 features, reduced to 79 after inputting them on npy_generator.py.
Also, am I correct to assume that 'timestep' in npy_generator refers to the number of windows? And that the value used should be (m − T + 1)*
*m = packets (rows) in the dataset
*T = number of packets in one window (set to 100 in the code from training/pi_queue.py)

**Reply by wobaolin** [reply]
Hello, do you have any processed datasets
